BJS2-3647 - Endpoints для контроллеров - Unicorn BC3

hhzktoeto opened the pull request.

hhzktoeto committed:
Endpoints для EventController + доработки

добавил эндпоинты для EventController
добавил доп.валидацию полей в UserDto
удалил пустой UserServiceTest

hhzktoeto committed:
Доработки в SkillController

hhzktoeto committed:
Мелкие исправления

hhzktoeto committed:
Изменил валидацию полей DTO

теперь валидация проходит с помощью аннотаций
удалил ненужные тесты/методы/импорты

IrinaVereshchagina13 commented on src/main/java/school/faang/user_service/controller/event/EventController.java:
```diff
@@ -1,47 +1,69 @@
 
-    public List<EventDto> getOwnedEvents(long userId) {
+    @Operation(summary = "Get events, which user with given ID owns")
+    @GetMapping("/user/{userId}/owned")
+    public List<EventDto> getOwnedEvents(@PathVariable long userId) {
         return eventService.getOwnedEvents(userId);
```
userId нужно получить через userContext

IrinaVereshchagina13 commented on src/main/java/school/faang/user_service/controller/event/EventController.java:
```diff
@@ -1,47 +1,69 @@
-    public List<EventDto> getParticipatedEvents(long userId) {
+    @Operation(summary = "Get events, which user with given ID participates in")
+    @GetMapping("/user/{userId}/participated")
+    public List<EventDto> getParticipatedEvents(@PathVariable long userId) {
```
Здесь тоже через userContext

hhzktoeto committed:
Исправления после review

в методах getOwnedEvents и getParticipatedEvents userId теперь берётся из UserContext'а
